### ISSUE TITLE
Filecount not working on Windows

### ISSUE THREAD

**Issue by Framnk** [influxdata/telegraf#5621]
### Relevant telegraf.conf:
```
[[inputs.filecount]]
  ## Directory to gather stats about.
  ##   deprecated in 1.9; use the directories option
  # directory = "/var/cache/apt/archives"

  ## Directories to gather stats about.
  ## This accept standard unit glob matching rules, but with the addition of
  ## ** as a "super asterisk". ie:
  ##   /var/log/**    -> recursively find all directories in /var/log and count files in each directories
  ##   /var/log/*/*   -> find all directories with a parent dir in /var/log and count files in each directories
  ##   /var/log       -> count all files in /var/log and all of its subdirectories
  directories = ["c:/temp"]

  ## Only count files that match the name pattern. Defaults to "*".
  name = "*"

  ## Count files in subdirectories. Defaults to true.
  recursive = false

  ## Only count regular files. Defaults to true.
  regular_only = true

  ## Only count files that are at least this size. If size is
  ## a negative number, only count files that are smaller than the
  ## absolute value of size. Acceptable units are B, KiB, MiB, KB, ...
  ## Without quotes and units, interpreted as size in bytes.
  size = "0B"

  ## Only count files that have not been touched for at least this
  ## duration. If mtime is negative, only count files that have been
  ## touched in this duration. Defaults to "0s".
  mtime = "0s"
```

### System info:

Telegraf 1.10.0~rc1 (git: HEAD 2d2abe29)
Windows 10 x64

### Steps to reproduce:

1. Include filecount plugin config with any Windows path specified
2. Run telegraf --debug --config <your config> --test

### Expected behavior:
Should see filecount counters in output

### Actual behavior:
Filecount counters never listed.

### Additional info:

Telegraf reports in log that filecount plugin was detected, just no output is ever shown.  Have tried several folders all containing files.

**Comment by danielnelson** [comment]
Reopening based on https://github.com/influxdata/telegraf/pull/5630#issuecomment-507683585

**Comment by glinton** [comment]
closed by #6077
